### PR TITLE
[cvss2] Implement cvss2 with types

### PR DESCRIPTION
**Pull request by n1chre**
The same implementation as for cvss3
Also delete ./cvss (old implementation)

### CODE REVIEW

File: cvss2/base_metrics.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	AvailabilityImpactPartial
+	AvailabilityImpactComplete
+)
+
+var (
+	weightsAvailabilityImpact = []float64{0, 0, 0.275, 0.66}
+	codeAvailabilityImpact    = []string{"", "N", "P", "C"}
+)
+
+func (ai AvailabilityImpact) defined() bool {
```
Comment (skogtwin) [inline cvss2/base_metrics.go]: Just heads up, not  a request for action, but it is idiomatic to keep receivers consistent, unless some other considerations were specifically taken: https://golang.org/doc/faq#methods_on_values_or_pointers
```
Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on method sets for details. 
```

File: cvss2/score.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+package cvss2
+
+import (
+	"fmt"
+	"math"
+)
+
+func roundTo1Decimal(x float64) float64 {
+	// round up to one decimal
```
Comment (skogtwin) [inline cvss2/score.go]: Feels like it belongs to above the function definition. Frankly, considering function's name, I'd dropped it whatsoever. A total nitpick, tho.

File: cvss2/vector.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	BaseMetrics
+	TemporalMetrics
+	EnvironmentalMetrics
+}
+
+// String returns this vectors representation as a string
+// it shouldn't depend on the order of metrics
+func (v Vector) String() string {
+	var sb strings.Builder
+	fmt.Fprint(&sb, "(")
```
Comment (skogtwin) [inline cvss2/vector.go]: So, you define `partSeparator` and `metricSeparator` but not `prefix` and `suffix`? :) Just nitpicking.

File: cvss2/vector.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	fmt.Fprint(&sb, ")")
+
+	return sb.String()
+}
+
+// VectorFromString will parse a string into a Vector, or return an error if it can't be parsed
+func VectorFromString(str string) (Vector, error) {
+	// remove parenthesis if they exists
```
Comment (skogtwin) [inline cvss2/vector.go]: seriosly, these comments...

File: cvss2/score.go (hunk 2 above)
Comment (n1chre) [reply cvss2/score.go]: dropped

File: cvss2/vector.go (hunk 3 above)
Comment (n1chre) [reply cvss2/vector.go]: added {pre,suf}fix

File: cvss2/vector.go (hunk 4 above)
Comment (n1chre) [reply cvss2/vector.go]: dropped

File: cvss2/base_metrics.go (hunk 1 above)
Comment (n1chre) [reply cvss2/base_metrics.go]: I won't change this for now. Maybe in the future for both cvss 2 and 3

**Comment by n1chre** [comment]
Shipping as per stamp

**Pull request merged**